sys/include/psa_crypto/psa/crypto_values.h

```diff
@@ -30,2028 +30,14 @@ extern "C" {
- *          - @ref PSA_KEY_DERIVATION_INPUT_SECRET is the PSK. The PSK must not be larger than
- *            @ref PSA_TLS12_PSK_TO_MS_PSK_MAX_SIZE.
- *          - @ref PSA_KEY_DERIVATION_INPUT_LABEL is the label.
- *
- *          Each input may only be passed once.
- *
- *          For the application to TLS-1.2:
- *          -   The seed, which is forwarded to the TLS-1.2 PRF, is the concatenation of the
- *              @c ClientHello.Random + @c ServerHello.Random.
- *          -   The label is "master secret" or "extended master secret".
- *
- *          @b Compatible @b key @b types
- *          - @ref PSA_KEY_TYPE_DERIVE (for the PSK)
- *          - @ref PSA_KEY_TYPE_RAW_DATA (for the other inputs)
- */
-#define PSA_ALG_TLS12_PSK_TO_MS(hash_alg) \
-    ((psa_algorithm_t)(0x08000300 | ((hash_alg) & 0x000000ff)))
-
-/**
- * @brief   Macro to build a truncated MAC algorithm.
- *
- * @details A truncated MAC algorithm is identical to the corresponding MAC algorithm except that
- *          the MAC value for the truncated algorithm consists of only the first mac_length bytes
- *          of the MAC value for the untruncated algorithm.
- *
- * @note    This macro might allow constructing algorithm identifiers that are not valid, either
- *          because the specified length is larger than the untruncated MAC or because the
- *          specified length is smaller than permitted by the implementation.
- *
- * @note    It is implementation-defined whether a truncated MAC that is truncated to the same
- *          length as the MAC of the untruncated algorithm is considered identical to the
- *          untruncated algorithm for policy comparison purposes.
- *
- *          The untruncated MAC algorithm can be recovered using @ref PSA_ALG_FULL_LENGTH_MAC().
- *
- *          @b Compatible @b key @b types
- *          The resulting truncated MAC algorithm is compatible with the same key types as the MAC
- *          algorithm used to construct it.
- *
- * @param   mac_alg     A MAC algorithm: a value of type @ref psa_algorithm_t such that
- *                      @ref PSA_ALG_IS_MAC(@p mac_alg) is true. This can be a truncated or
- *                      untruncated MAC algorithm.
- * @param   mac_length  Desired length of the truncated MAC in bytes. This must be at most the
- *                      untruncated length of the MAC and must be at least an
- *                      implementation-specified minimum. The implementation-specified minimum must
- *                      not be zero.
- *
- * @return  The corresponding MAC algorithm with the specified length.
- *          Unspecified if @c mac_alg is not a supported MAC algorithm or if @c mac_length is too
- *          small or too large for the specified MAC algorithm.
- */
-#define PSA_ALG_TRUNCATED_MAC(mac_alg, mac_length) \
-    ((psa_algorithm_t)(((mac_alg) & ~0x003f0000) | (((mac_length) & 0x3f) << 16)))
+#include "psa/hash/algorithm.h"
```

Comment:
This way all of these are always included. Since this is about not including everything, it might be better to make them conditional. E.g.
```
#if IS_USED(MODULE_PSA_HASHES)
#include "psa/hash/algorithm.h"
#endif
```

Reply:
Ok

Reply:
I think this would force users who need to use a single algorithm definition to pull-in the whole dispatcher code of the affected modules.

For example users who want to use hash algorithm definitions for defining HMAC algorithms would be forced to include the whole dispatcher code for the hashes too.

Reply:
So i would like to keep it as it is.